lib/spack/llnl/util/filesystem.py

```diff
@@ -884,7 +898,10 @@ def mkdirp(*paths, **kwargs):
                         os.chmod(intermediate_path, intermediate_mode)
                     if intermediate_group is not None:
                         chgrp_if_not_world_writable(intermediate_path, intermediate_group)
-                        os.chmod(intermediate_path, intermediate_mode)  # reset sticky bit after
+                        if intermediate_mode is not None:
```

Comment:
Without this check `mypy` was (correctly) warning about possibly calling `os.chmod` with `intermediate_mode` set to `None`

Reply:
@becker33 fyi, so you can double check if this the correct fix here